python_modules/dagster/dagster/core/asset_defs/assets_job.py

```diff
@@ -110,22 +122,26 @@ def asset2(asset1):
                 )
             all_resource_defs[resource_key] = resource_def
 
-    # turn any AssetsDefinitions into SourceAssets
-    resolved_source_assets: List[SourceAsset] = []
-    for asset in source_assets or []:
-        if isinstance(asset, AssetsDefinition):
-            resolved_source_assets += asset.to_source_assets()
-        elif isinstance(asset, SourceAsset):
-            resolved_source_assets.append(asset)
+    required_io_manager_keys = set()
+    for source_asset in resolved_source_assets:
+        if not source_asset.io_manager_def:
+            required_io_manager_keys.add(source_asset.get_io_manager_key())
+        else:
+            all_resource_defs[source_asset.get_io_manager_key()] = source_asset.io_manager_def
+
+    all_resource_defs = merge_dicts({"io_manager": fs_asset_io_manager}, all_resource_defs)
```

Comment:
I think there may be a merge conflict here, because we just removed fs_asset_io_manager